### ISSUE TITLE
Some discuss about replaceAll() work with nested routes

### ISSUE THREAD

**Issue by LonlyCat** [Milad-Akarie/auto_route_library#1589]
Let's say we have this structure:

```
@AutoRouterConfig()
class AppRouter extends _$AppRouter {
  @override
  List<AutoRoute> get routes => [
        CustomRoute(
            initial: true,
            path: '/',
            page: AutoMainRouteScaffold.page,
            transitionsBuilder: TransitionsBuilders.fadeIn,
            children: [
              AutoRoute(
                path: 'home/:category',
                page: HomeRoute.page,
              ),
              AutoRoute(
                path: 'shoppingCart',
                page: ShoppingCartRoute.page,
                guards: [AuthGuard()],
              ),
              AutoRoute(path: 'mine', page: MineRoute.page),
            ]),
        CustomRoute(
            path: '/product/:id',
            page: ProductDetailRoute.page,
            transitionsBuilder: TransitionsBuilders.fadeIn),
        CustomRoute(
            path: '/login',
            page: LoginRoute.page,
            transitionsBuilder: TransitionsBuilders.fadeIn),
      ];
}
```
now we're in the following location `/mine`, and we want to go location like this `/shoppingCart/product/2`.
A ideal use case `context.router.root.replaceAll([ShoppingCartRoute(), ProductDetailRoute(productId: 1)])`. It does push the new route onto the stack, but the nesting route is not in the right index.

When I debug `routing_controller.dart`, `StackRouter.replaceAll()` simply pushes *routes* onto the stack, The `activeIndex` that might be the TabRouter is not updated.

I tried to modify the function `replaceAll()`

```
// Before
Future<void> replaceAll(
    List<PageRouteInfo> routes, {
    OnNavigationFailure? onFailure,
  }) {
    
    final scope = _findStackScope(routes.first);
    scope._pages.clear();
    markUrlStateForReplace();
    return scope._pushAll(routes, onFailure: onFailure);
    final matches = _matchAllOrReportFailure(routes, onFailure);
  }

// After
Future<void> replaceAll(
    List<PageRouteInfo> routes, {
    OnNavigationFailure? onFailure,
  }) {
    
    final scope = _findStackScope(routes.first);
    scope._pages.clear();
    markUrlStateForReplace();
    Future<void> _completed = scope._pushAll(routes, onFailure: onFailure);
    final matches = _matchAllOrReportFailure(routes, onFailure);
    if (matches != null) {
      final mayUpdateRoute = matches.first;
      final mayUpdateController = _innerControllerOf(mayUpdateRoute.key);
      if (mayUpdateController != null) {
        final newChildren = mayUpdateRoute.children ?? const [];
        if (mayUpdateController.managedByWidget) {
          mayUpdateController._onNavigate(newChildren);
        }
        return mayUpdateController._navigateAll(
          newChildren,
          onFailure: onFailure,
        );
      }
    }
    return _completed;
  }
```

and now try call `context.router.root.replaceAll([ShoppingCartRoute(), ProductDetailRoute(productId: 1)])` again, The `ShoppingCartRoute` in nested routes will be display correct.

But I'm not sure why didn't do that before, can someone explain that to me?

**Comment by Milad-Akarie** [comment]
@LonlyCat ya actually I've been working on replaceAll because of a similar issue,  it was fixed in version 7.4.0, also added flag for ditching old states
replaceAll([Routes], updateExistingEntries: false)

**Comment by LonlyCat** [comment]
That sounds good! It works good for me.

> @LonlyCat ya actually I've been working on replaceAll because of a similar issue, it was fixed in version 7.4.0, also added flag for ditching old states replaceAll([Routes], updateExistingRoutes: false)
